docs/reference/contributing/target/target.md

```diff
@@ -34,9 +70,173 @@ Add the target description to `mbed-os\targets\targets.json` using keys that the
+
+Define a default for each peripheral. Define the DAPLink serial pins as `STDIO_UART_RX` and `STDIO_UART_TX`. For example:
+
+```C
+#ifndef MBED_PINNAMES_H
+#define MBED_PINNAMES_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+    PA_0  = 0x00,
+    PA_1  = 0x01,
+    PA_1  = 0x01,
+<snip>
+
+    // Generic signals namings
+    LED1        = PB_2,
+    LED2        = PB_10,
+    LED_RED     = LED1,
+    LED_BLUE    = LED2,
+    USER_BUTTON = PC_13,
+
+    // Standardized button names
+    BUTTON1 = USER_BUTTON,
+
+    SERIAL_TX   = PA_8,
+    SERIAL_RX   = PA_12,
+
+    I2C_SCL     = PA_17,
+    I2C_SDA     = PA_18,
+
+    SPI_MOSI    = PA_4,
+    SPI_MISO    = PA_7,
+    SPI_SCK     = PA_6,
+    SPI_CS      = PA_16,
+
+    //DAPLink
+    STDIO_UART_RX = SERIAL_RX
+    STDIO_UART_TX = SERIAL_TX
+    USBRX      = STDIO_UART_RX,
+    USBTX      = STDIO_UART_TX,
+    SWDIO      = PA_26,
+    SWCLK      = PA_25,
+    NTRST      = PA_13,
+} PinName;
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
+```
+
+### Compiling with a supported compiler
+
+You can verify that the new board port compiles by using an [example application](/docs/development/tutorials/mbed-os-quick-start.html), and checking out the branch containing your port in the `mbed-os` subdirectory.
+
+### Sample directory structure
+
+Below is an example of all of the files that you would add for a target from vendor `MY_VENDOR`, with an MCU `MY_DEVICE_1`, from device family `MY_FAMILY_1`, mounted on the board `MY_BOARD_1` and supporting the `SERIAL` device:
+
+```
+mbed-os
+└── targets
+   └── TARGET_MY_VENDOR
+      ├── TARGET_MY_FAMILY
+      │  ├── TARGET_MY_DEVICE_1
+      │  │  ├── TARGET_MY_BOARD_1
+      │  │  │  ├── PeripheralNames.h
+      │  │  │  ├── PeripheralPins.c
+      │  │  │  ├── PeripheralPins.h
+      │  │  │  ├── clock_config.c
+      │  │  │  ├── clock_config.h
+      │  │  │  └── device.h
+      │  │  ├── TOOLCHAIN_ARM
+      │  │  │  ├── device.sct
+      │  │  │  └── startup.S
+      │  │  ├── TOOLCHAIN_GCC
+      │  │  │  ├── device.ld
+      │  │  │  └── startup.S
+      │  │  ├── TOOLCHAIN_IAR
+      │  │  │  ├── device.icf
+      │  │  │  └── startup.S
+      │  │  ├── clock.c
+      │  │  ├── clock.h
+      │  │  ├── uart.c
+      │  │  ├── uart.h
+      │  │  └── us_ticker.c
+      │  └── api
+      │     ├── Objects.h
+      │     └── PortNames.h
+      └── mbed_rtx.h
+```
+
+### HAL porting
 
-There are many more APIs to implement. You enable the following APIs by adding a `device_has` attribute to the MCU_NAME target definition in `targets.json` and providing an implementation of the API declared in the API header.
+There are many more APIs to implement. You enable the following APIs by adding a `device_has` attribute to the `MCU_NAME` target definition in `targets.json`, and providing an implementation of the API declared in the API header.
 
 device_has       |   API header
 -----------------|------------------
```

Comment:
It would be useful to create a link for each of the below header files to actual files in mbed-os repo. Not a big deal, but would be convenient.

Reply:
We could at least make them a link to their respective porting pages.